components/cosmetic_filters/resources/data/element_picker.ts

```diff
@@ -542,6 +554,267 @@ const elementPickerUserModifiedRule = (selector: string) => {
+  if (!element) return;
+
+  const min = parseInt(element.dataset.min ?? '0', 10);
+  const max = parseInt(element.dataset.max ?? '4', 10);
+  const initialValue = 4
+
+  const thumb = createThumbElement();
+  element.appendChild(thumb);
+
+  element.tabIndex = 0;
+
+  let isDragging = false;
+  let currentValue = initialValue;
+
+  const updateSlider = (value: number, fireEvent = true): number => {
+    value = Math.max(min, Math.min(max, value));
+
+    const percentage = ((value - min) / (max - min)) * 100;
+
+    // Update thumb position and fill
+    thumb.style.left = `${percentage}%`;
+    element.style.setProperty('--fill-percentage', `${percentage}%`);
+    element.style.setProperty('--value', value.toString());
+
+    // Set track fill width
+    element.style.setProperty('--track-fill-width', `${percentage}%`);
+    element.style.setProperty('--before-width', `${percentage}%`);
+
+    currentValue = value;
+
+    if (fireEvent && options.onChange) {
+      options.onChange(value);
+    }
+    return value;
+  };
+
+  element.style.setProperty(
+    '--before-width', `${((currentValue - min) / (max - min)) * 100}%`);
+
+  const handleMove = (e: MouseEvent | TouchEvent) => {
+    if (!isDragging && e.type !== 'click') return;
+
+    e.preventDefault();
+
+    const rect = element.getBoundingClientRect();
+    const clientX = e.type.includes('touch')
+      ? (e as TouchEvent).touches[0].clientX
+      : (e as MouseEvent).clientX;
+
+    // Calculate new value based on pointer position
+    const percentage = Math.max(
+      0, Math.min(1, (clientX - rect.left) / rect.width));
+    const newValue = Math.round(min + percentage * (max - min));
+
+    updateSlider(newValue);
+  };
+
+  function throttle<T extends (e: MouseEvent | TouchEvent) => any>(
+    func: T,
+    delay: number
+  ): (this: ThisParameterType<T>, e: MouseEvent | TouchEvent) => ReturnType<T> {
+    let timeoutId: ReturnType<typeof setTimeout> | null = null;
+    let lastArgs: [MouseEvent | TouchEvent] | null = null;
+    let lastThis: ThisParameterType<T> | null = null;
+    let lastResult: ReturnType<T> | null = null;
+
+    return function (this: ThisParameterType<T>, e: MouseEvent | TouchEvent)
+      : ReturnType<T> {
+      lastArgs = [e];
+      lastThis = this;
+      if (!timeoutId) {
+        lastResult = func.apply(lastThis, lastArgs);
+        timeoutId = setTimeout(() => {
+          timeoutId = null;
+          if (lastArgs !== null) {
+            setTimeout(() => {
+              const currentArgs = lastArgs!;
+              const currentThis = lastThis;
+              lastArgs = null;
+              lastThis = null;
+              func.apply(currentThis, currentArgs);
+            }, 0);
+          }
+        }, delay);
+      }
+      if (lastResult !== null) {
+        return lastResult;
+      }
+      return func.apply(this, [e]);
+    };
+  }
+
+  const throttledHandleMove = throttle(handleMove, 50);
```

Comment:
do we actually want to throttle the moving? It feels like instant feedback here is probably a good thing